pkg/common/upgrade/managed.go

```diff
@@ -483,3 +471,112 @@ func isUpgradeTriggered(h *helper.H, desired *configv1.Update) (bool, error) {
 
 	return false, nil
 }
+
+// create cluster workloads prior to upgrade commencement which will test
+// manage-upgrade-operator's drain strategy functions
+func createUpgradeClusterWorkloads(h *helper.H) error {
+
+	// Create Pod Disruption Budget test workloads if desired
+	if viper.GetBool(config.Upgrade.ManagedUpgradeTestPodDisruptionBudgets) {
+		pdbPodPrefixes := []string{"pdb"}
+		err := createManagedUpgradeWorkload(pdbWorkloadName, pdbWorkloadDir, pdbPodPrefixes, h)
+		if err != nil {
+			return fmt.Errorf("unable to setup PDB workload for upgrade: %v", err)
+		}
+	}
+
+	// Create Node Drain test workloads if desired
+	if viper.GetBool(config.Upgrade.ManagedUpgradeTestNodeDrain) {
+		drainPodPrefixes := []string{"node-drain-test"}
+		err := createManagedUpgradeWorkload(drainWorkloadName, drainWorkloadDir, drainPodPrefixes, h)
+		if err != nil {
+			return fmt.Errorf("unable to setup node drain test workload for upgrade: %v", err)
+		}
+	}
+
+	return nil
+}
+
+// determine which policy provider to use for an upgrade policy source
+func getPolicyProvider() string {
```

Comment:
could this be named `getProviderSource`?

Reply:
Makes sense - function now renamed.